pkg/skaffold/initializer/build/init.go

```diff
@@ -62,6 +63,16 @@ func (d *defaultBuildInitializer) PrintAnalysis(out io.Writer) error {
 	return printAnalysis(out, d.enableNewFormat, d.skipBuild, d.builderImagePairs, d.builders, d.unresolvedImages)
 }
 
+func (d *defaultBuildInitializer) UnresolvedPairs() []GeneratedBuilderImagePair {
+	return d.generatedBuilderImagePairs
+}
+
+func (d *defaultBuildInitializer) Resolve() {
```

Comment:
I am not sure if i understand this correctly. The `UnresolvedPairs` i.e. `d.generatedBuilderImagePairs` is appended in this function to `d.builderImagePairs`. 
There is no processing that happens here. 
Shd this just be called append?

Reply:
the processing happens when the manifests are generated - this is a function for the controller to call, to tell the initializer that all builder/image pairs have been processed. in practice it does only do appending, but `Resolve` is more descriptive.